aiokafka/protocol/metadata.py

```diff
@@ -238,23 +242,23 @@ class MetadataRequest_v5(Request):
     API_VERSION = 5
     RESPONSE_TYPE = MetadataResponse_v5
     SCHEMA = MetadataRequest_v4.SCHEMA
-    ALL_TOPICS = -1  # Null Array (len -1) for topics returns all topics
-    NO_TOPICS = None  # Empty array (len 0) for topics returns no topics
+    ALL_TOPICS: Optional[int] = -1  # Null Array (len -1) for topics returns all topics
+    NO_TOPICS: None = None  # Empty array (len 0) for topics returns no topics
 
 
-MetadataRequest = [
+MetadataRequest = (
```

Comment:
Do I understand correctly that the aim of this change is to solve problem with variance? Semantically tuple doesn't fit well here.

Reply:
I think the first variant is more correct.

```python
MetadataRequestTuple = (
    MetadataRequest_v0,
    MetadataRequest_v1,
    MetadataRequest_v2,
    MetadataRequest_v3,
    MetadataRequest_v4,
    MetadataRequest_v5,
)

reveal_type(MetadataRequestTuple)  # Revealed type is "Tuple[def (*args: Any, **kwargs: Any) -> aiokafka.protocol.metadata.MetadataRequest_v0, def (*args: Any, **kwargs: Any) -> aiokafka.protocol.metadata.MetadataRequest_v1, def (*args: Any, **kwargs: Any) -> aiokafka.protocol.metadata.MetadataRequest_v2, def (*args: Any, **kwargs: Any) -> aiokafka.protocol.metadata.MetadataRequest_v3, def (*args: Any, **kwargs: Any) -> aiokafka.protocol.metadata.MetadataRequest_v4, def (*args: Any, **kwargs: Any) -> aiokafka.protocol.metadata.MetadataRequest_v5]"
reveal_type(MetadataRequestTuple[0])  # Revealed type is "def (*args: Any, **kwargs: Any) -> aiokafka.protocol.metadata.MetadataRequest_v0"

MetadataRequestList = [
    MetadataRequest_v0,
    MetadataRequest_v1,
    MetadataRequest_v2,
    MetadataRequest_v3,
    MetadataRequest_v4,
    MetadataRequest_v5,
]

reveal_type(MetadataRequestList)  # Revealed type is "builtins.list[def (*args: Any, **kwargs: Any) -> aiokafka.protocol.api.Request]"
reveal_type(MetadataRequestList[0])  # Revealed type is "def (*args: Any, **kwargs: Any) -> aiokafka.protocol.api.Request"
```

Reply:
The semantic of tuple is: a fixed number of fields each having it's own type, i.e. like struct, but with access by index instead of name. The semantic of list is: a variable array of items with the same type. Here, we have an array of items each being a request type.

Reply:
But in some cases we need to know type of "field", for example (aiokafka/client.py):

```python
        if self.api_version > (0, 11):
            request = FindCoordinatorRequest[1](coordinator_key, coordinator_type)
        else:
            # Group coordination only
            assert (
                coordinator_type == CoordinationType.GROUP
            ), "No transactions for older brokers"
            request = FindCoordinatorRequest[0](coordinator_key)
```

Reply:
Do you see any reason of using `FindCoordinatorRequest[1]` instead of `FindCoordinatorRequest_v1` there? To me the variant with index is somewhat less reliable way to refer to particular version of request.

Reply:
You are right, but it's just example from aiokafka :)

Ok, I reverted my changes